src/components/instructor/AssignmentsView.js

```diff
@@ -10,15 +16,133 @@ import {useLocation} from 'react-router-dom'
+        }
+    }
+
+    useEffect( () => {
+        fetchAsgnmts();
+    }, []);
+
+    const gradeAsgnmt = async (asgnmt) => {
+        try {
+            const response = await fetch(`${SERVER_URL}/grades`,
+                {
+                    method: 'PUT',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify(asgnmt),
+                });
+            if (response.ok) {
+                setMessage("Grade Saved")
+                fetchAsgnmts();
+            } else {
+                const rc = await response.json();
+                setMessage(rc.message);
+            }
+        } catch (err) {
+            setMessage("network error: "+err);
+        }
+    }
+
+    const editChange = (event) => {
+        setSearch({...search,  [event.target.name]:event.target.value});
+    }
+    const deleteAsgnmt = async (id) => {
+        try {
+            const response = await fetch (`${SERVER_URL}/assignments/${id}`,
+                {
+                    method: 'DELETE',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                });
+            if (response.ok) {
+                setMessage("Assignment deleted");
+                fetchAsgnmts();
+            } else {
+                const rc = await response.json();
+                setMessage(rc.message);
+            }
+        } catch (err) {
+            setMessage("network error: "+err);
+        }
+    }
+
+    const onDelete = (e) => {
+        const row_idx = e.target.parentNode.parentNode.rowIndex - 1;
+        const id = asgnmts[row_idx].id;
+        confirmAlert({
+            title: 'Confirm to delete',
+            message: 'Do you really want to delete?',
+            buttons: [
+                {
+                    label: 'Yes',
+                    onClick: () => deleteAsgnmt(id)
+                },
+                {
+                    label: 'No',
+                }
+            ]
+        });
+    }
+
+    return (
+        <>
+            <div>
+                <h3>Assignments</h3>
+                <h4>{message}</h4>
+                <table className="Center" >
+                    <thead>
+                        <tr>
+                            {headers.map((s, idx) => (<th key={idx}>{s}</th>))}
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {asgnmts.map((a) => (
+                            <tr key={a.id}>
+                                <td>{a.id}</td>
+                                <td>{a.title}</td>
+                                <td>{a.dueDate}</td>
+                                <td>{a.courseId}</td>
+                                <td>{a.secId}</td>
+                                <td>{a.secNo}</td>
+                                <td><AssignmentUpdate asgnmts={a} onClose={gradeAsgnmt()} /></td>
```

Comment:
`onClose` need to be `fetchAsgnmts`. `onClose={fetchAsgnmts}`. Then in assignmentUpdate.js you need to add `props.onClose();` to `editClose()` function